### ISSUE TITLE
Technically not an issue

### ISSUE THREAD

**Issue by polfosol** [ubiqsecurity/ubiq-fpe-c#1]
Hi

I just wanted to let you know that I have used some of your test vectors to verify my C-implementation of the FPE algorithms. Of course I have mentioned the source and credit. Please feel free to check out the `testvectors` folder of [this repository](https://github.com/polfosol/micro-AES) and inform me of any issues.

Now there is a problem that I wasn't able to figure out. The NIST has provided some sample vectors in [this page](https://pages.nist.gov/ACVP/draft-celi-acvp-symmetric.html). I could verify all of them using my own code, except two FF1 vectors that use 128-bit tweaks. You can find them at the bottom of the page:
```
"tcId": 2,
"tweak": "CB81CF732B22A983B2C6E584726C9F59",
"tweakLen": 128,
"pt": "60454384602180796680544707896451618557756152702734587161",
"key": "E3EFDAF1ABEA7863A0A95F833420D083"

"tcId": 27,
"tweak": "994C168B9F6225C4BC12A83561C0E1A6",
"tweakLen": 128,
"pt": "uxmdsdjbsywthsvzjlfcwlmpkarnaeoirtihgfuu",
"key": "53CA14AF6F97612C96FFAA2BA8D88C44"
```
This seems ironic since I have exactly followed the guidelines in SP-800-38G, and your library **_fails to verify them_** too. I double-checked and triple-checked the code with no luck. So I hope you guys @garyschneir @johntyner have some ideas about this.

Thanks

**Comment by johntyner** [comment]
Hey @polfosol, thanks for pointing this out. I apologize for the long delay in investigating this. That said, I ended up corresponding with the authors of the draft that you linked to. Apparently, that draft was created using vectors that "were generated when the original code was developed and may have since been bugfixed." They pointed me instead at vectors listed [here](https://github.com/usnistgov/ACVP-Server/tree/master/gen-val/json-files/ACVP-AES-FF1-1.0). I pulled a random sampling of those vectors and ran them through both our C and Python implementations, and they all verified successfully.
